ctools/defragment_sharded_collection.py

```diff
@@ -871,6 +871,18 @@ async def split_oversized_chunks(shard, progress):
             break
         
         pass # while max_iterations > 0:
+       
+    if args.exec_phase == 'phase2' or args.exec_phase == 'all':
```

Comment:
technically we have the size there so we should not need to recalculate it

Reply:
The chunk that loose the size is the one with the highest version on the donor, this means that it could happen that you are merging `C1` + `C2` and the internal migration reset the size of `C3`.
In addition we don't really care if size estimation gets wiped out from `config.chunks` while running phase 2 in fact during this phase execution we just rely on the in-memory representation of chunks.
The reason for which I added this additional re-calculation step here is to ensure all the size estimation are persisted before to execute phase3.

Regarding your second comment I guess you are right, we could re-use the already estimated size that we have in memory, but for the moment I would first fix the bug and then we can think on how to  improve the performance.